blockmanager.go

```diff
@@ -309,10 +307,7 @@ func (b *blockManager) startSync(peers *list.List) {
 		// and fully validate them.  Finally, regression test mode does
 		// not support the headers-first approach so do normal block
 		// downloads when in regression test mode.
-		if b.nextCheckpoint != nil &&
-			best.Height < b.nextCheckpoint.Height &&
-			!cfg.RegressionTest && !cfg.DisableCheckpoints {
```

Comment:
If either `cfg.RegressionTest` or `cfg.DisableCheckpoints`, then `b.nextCheckpoint` is guaranteed to be `nil`, so this check is redundant.